### PR TITLE
Attach image

### PR DESCRIPTION
**Pull request by gsim1903**

### CODE REVIEW

**Comment by gsim1903** [comment]
https://www.pivotaltracker.com/story/show/182082481

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -8,16 +8,22 @@ def index
...
+    render json: { article: }
   end
 
   def create
-    new_article = Article.create(params[:article].permit!)
+    category = Category.find_by(name: params[:article][:category])
+    new_article = Article.new(article_params)
+    new_article.category = category
+    new_article.save
+    new_article.attach_image(params[:article][:image])
```
Comment (miyeaier) [inline app/controllers/api/articles_controller.rb:19]: on line 16 new method has been used why you didn't use create method

File: app/models/article.rb [hunk 2]
```diff
@@ -1,3 +1,28 @@
 class Article < ApplicationRecord
   belongs_to :category
+  has_one_attached :image
+
```
Comment (miyeaier) [inline app/models/article.rb:4]: why do we just have 1 image, an article can have many images

**Review by miyeaier — APPROVED** [review]

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (shagufta333) [reply app/controllers/api/articles_controller.rb:19]: i think we could use create method from ruby ibrary to create articles and to  save it directly to database.

File: app/controllers/api/articles_controller.rb [hunk 3]
```diff
@@ -29,5 +35,4 @@ def serialize_categories(categories)
     end
     response
```
Comment (shagufta333) [inline app/controllers/api/articles_controller.rb:36]: i did not understand the placement of 'response' here

File: spec/models/article_spec.rb [hunk 4]
```diff
@@ -1,19 +1,31 @@
...
   end
-  
+
   describe 'Factory' do
     it 'is expected to be valid' do
       expect(create(:article)).to be_valid
     end
   end
 
-end
```
Comment (shagufta333) [inline spec/models/article_spec.rb:19]: lots of new codes and methods. which haven't seen before.

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (gsim1903) [reply app/controllers/api/articles_controller.rb:19]: @maxarvid went through some de-bugging onthis. I understand we are defining the create method i.e. what happens when we wish to create an article. 

So we are saying it needs to find a category contain params and and image can be saved onto the record). Least thats what I think!

File: app/models/article.rb (hunk 2 above)
Comment (gsim1903) [reply app/models/article.rb:4]: Maybe we add this as an extra feature in the PT

File: spec/models/article_spec.rb (hunk 4 above)
Comment (gsim1903) [reply spec/models/article_spec.rb:19]: From CA video re file attachments.

**Pull request merged**